xcube_cds/store.py

```diff
@@ -699,40 +700,54 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
         )
 
     @classmethod
-    def get_type_ids(cls) -> Tuple[str, ...]:
-        return TYPE_ID_DATASET,
+    def get_type_specifiers(cls) -> Tuple[str, ...]:
+        return TYPE_SPECIFIER_CUBE,
 
-    def get_data_ids(self, type_id: Optional[str] = None) -> \
-            Iterator[Tuple[str, Optional[str]]]:
-        self._assert_valid_type_id(type_id)
-        return iter((data_id,
-                     self._handler_registry[data_id].
-                     get_human_readable_data_id(data_id))
-                    for data_id in self._handler_registry)
+    def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
+        self._validate_data_id(data_id)
+        return TYPE_SPECIFIER_CUBE,
+
+    def get_data_ids(self, type_specifier: Optional[str] = None,
+                     include_titles: bool = True)\
+            -> Iterator[Tuple[str, Optional[str]]]:
+        if not self._is_type_specifier_compatible(type_specifier):
+            # If the type specifier isn't compatible, return an empty iterator.
+            return iter(())
+        return iter(
+            (data_id,
+             self._handler_registry[data_id].
+                get_human_readable_data_id(data_id) if include_titles else None)
+            for data_id in self._handler_registry
+        )
 
-    def has_data(self, data_id: str) -> bool:
-        return data_id in self._handler_registry
+    def has_data(self, data_id: str, type_specifier: Optional[str] = None)\
+            -> bool:
+        return self._is_type_specifier_compatible(type_specifier) and\
+            data_id in self._handler_registry
 
-    def describe_data(self, data_id: str) -> DataDescriptor:
+    def describe_data(self, data_id: str, type_specifier: Optional[str] = None)\
+            -> DataDescriptor:
         self._validate_data_id(data_id)
+        self._validate_type_specifier(type_specifier)
         return self._handler_registry[data_id].describe_data(data_id)
 
     # noinspection PyTypeChecker
-    def search_data(self, type_id: Optional[str] = None, **search_params) -> \
-            Iterator[DataDescriptor]:
-        self._assert_valid_type_id(type_id)
+    def search_data(self, type_specifier: Optional[str] = None,
+                    **search_params)\
+            -> Iterator[DataDescriptor]:
+        self._validate_type_specifier(type_specifier)
         raise NotImplementedError()
 
     def get_data_opener_ids(self, data_id: Optional[str] = None,
-                            type_id: Optional[str] = None) -> \
-            Tuple[str, ...]:
-        self._assert_valid_type_id(type_id)
+                            type_specifier: Optional[str] = None)\
+            -> Tuple[str, ...]:
+        self._validate_type_specifier(type_specifier)
         self._assert_valid_opener_id(data_id)
```

Comment:
This is not part of the pull request, but this line does not make sense. I guess it should be self._validate_data_id(data_id) .

Reply:
Thanks, fixed.